Sources/SignalProducer.swift

```diff
@@ -1817,6 +1817,39 @@ extension SignalProducer {
 		}
 	}
 
+	/// Delays retrying on failure by `interval` up to `count` attempts.
+	///
+	/// - precondition: `count` must be non-negative integer.
+	///
+	/// - parameters:
+	///   - count: Number of retries.
+	///   - interval: An interval between invocations.
+	///   - scheduler: A scheduler to deliver events on.
+	///
+	/// - returns: A signal producer that restarts up to `count` times.
+	public func retry(upTo count: Int, interval: TimeInterval, on scheduler: DateScheduler) -> SignalProducer<Value, Error> {
+		precondition(count >= 0)
+		
+		if count == 0 {
+			return producer
+		}
+		
+		var retries = count
+		
+		return flatMapError { error in
+			// The final attempt shouldn't defer the error if it fails
+			var producer = SignalProducer<Value, Error>(error: error)
+			if retries > 0 {
+				producer = SignalProducer.empty
+					.delay(interval, on: scheduler)
+					.concat(producer)
+			}
+			
+			retries -= 1
+			return producer
+		}.retry(upTo: count)
```

Comment:
I don't believe this is correct. I think the `.retry` should be inside the `flatMapError`.

If you add a test like I suggested below, this should become clear.